bin/migrate-oats-data/submissions/app_submissions.py

```diff
@@ -0,0 +1,262 @@
+    cursor (obj): Cursor object to execute queries.
+    rows (list): Rows of data to update in the database.
+
+    Returns:
+    None: Commits the changes to the database.
+    """
+    (
+        nfu_data_list,
+        nar_data_list,
+        other_data_list,
+        exc_data_list,
+        inc_data_list,
+    ) = prepare_app_sub_data(rows)
+
+    if len(nfu_data_list) > 0:
+        execute_batch(
+            cursor,
+            get_update_query_for_nfu(),
+            nfu_data_list,
+            page_size=batch_size,
+        )
+
+    if len(nar_data_list) > 0:
+        execute_batch(
+            cursor,
+            get_update_query_for_nar(),
+            nar_data_list,
+            page_size=batch_size,
+        )
+
+    if len(exc_data_list) > 0:
+        execute_batch(
+            cursor,
+            get_update_query_for_exc(),
+            exc_data_list,
+            page_size=batch_size,
+        )
+
+    if len(inc_data_list) > 0:
+        execute_batch(
+            cursor,
+            get_update_query_for_inc(),
+            inc_data_list,
+            page_size=batch_size,
+        )
+
+    if len(other_data_list) > 0:
+        execute_batch(
+            cursor,
+            get_update_query_for_other(),
+            other_data_list,
+            page_size=batch_size,
+        )
+
+    conn.commit()
+
+def prepare_app_sub_data(app_sub_raw_data_list):
+    """
+    This function prepares different lists of data based on the 'alr_change_code' field of each data dict in 'app_sub_raw_data_list'.
+
+    :param app_sub_raw_data_list: A list of raw data dictionaries.
+    :return: Five lists, each containing dictionaries from 'app_sub_raw_data_list' grouped based on the 'alr_change_code' field
+
+    Detailed Workflow:
+    - Initializes empty lists
+    - Iterates over 'app_sub_raw_data_list'
+        - Maps the basic fields of the data dictionary based on the alr_change_code
+    - Returns the mapped lists
+    """
+    nfu_data_list = []
+    nar_data_list = []
+    exc_data_list = []
+    inc_data_list = []
+    other_data_list = []
+
+    for row in app_sub_raw_data_list:
+        data = dict(row)
+        # data = map_basic_field(data)
+
+        if data["alr_change_code"] == ALRChangeCode.NFU.value:
+            # data = mapOatsToAlcsAppPrep(data)
+            nfu_data_list.append(data)
+        elif data["alr_change_code"] == ALRChangeCode.NAR.value:
+            nar_data_list.append(data)
+        elif data["alr_change_code"] == ALRChangeCode.EXC.value:
+            # data = mapOatsToAlcsLegislationCode(data)
+            exc_data_list.append(data)
+        elif data["alr_change_code"] == ALRChangeCode.INC.value:
+            # data = mapOatsToAlcsLegislationCode(data)
+            inc_data_list.append(data)
+        else:
+            other_data_list.append(data)
+
+    return nfu_data_list, nar_data_list, other_data_list, exc_data_list, inc_data_list
+
+
+def get_update_query(unique_fields,unique_values):
```

Comment:
rename to get_insert_query or get_copy_query based on whatever query you will end up with after refactoring